### PR TITLE
Detecting CPU features to verify VM migration (jsc#SLE-10184)

### PR DESCRIPTION
**Pull request by tbazant**
### Description
The new procedure helps identify CPU features on VMhosts to use only those which are supported  on all of them during migraiton.

### CODE REVIEW

File: xml/xen_administrating.xml [hunk 1]
```diff
@@ -236,6 +237,77 @@
    </listitem>
   </itemizedlist>
 
+  <sect2 xml:id="sec-xen-manage-migrate-cpu">
+   <title>Detect CPU Features</title>
```
Comment (taroth21) [inline xml/xen_administrating.xml:241]:
```suggestion
   <title>Detecting CPU Features</title>
```

File: xml/xen_administrating.xml [hunk 2]
```diff
@@ -236,6 +237,77 @@
...
+    "0x00000007,0x00:ebx=xxxxxxxxxxxxxxxxxx00x0000x0x0x00"
+]
+</screen>
+    </step>
+    <step>
+     <para>
+      Copy the output <literal>cpuid=[...]</literal> configuration snipped into
+      the <command>xl</command> configuration of the migrated
+      guest<filename>domU.cfg</filename> or alternatively to its &libvirt;'s
+      XML configuration.
```
Comment (jfehlig) [inline xml/xen_administrating.xml:286]: I'm not familiar with this tool, but is it really capable of producing libvirt <cpu> config from cpuid=[...] in xl.cfg? I guess we'll need some input from Olaf Hering.

**Comment by jfehlig** [comment]
BTW, libvirt also supports calculating a baseline CPU for migration. It is already documented in the best practices guide. See "4.5.2.3 Virtual CPU Migration Considerations"

https://documentation.suse.com/sles/15-SP1/single-html/SLES-vt-best-practices/

File: xml/xen_administrating.xml (hunk 2 above)
Comment (olafhering) [reply xml/xen_administrating.xml:286]: Yes, it produces also XML snippets. 

https://github.com/olafhering/xen_maskcalc/blob/9f48802ef81dd2370036e97897ab19b9cb15ca68/xen_maskcalc.py#L262

File: xml/xen_administrating.xml [hunk 3]
```diff
@@ -236,6 +237,77 @@
...
+    </step>
+    <step>
+     <para>
+      Start the source guest with the <emphasis>trimmed</emphasis> CPU
+      configuration. The guest can now only use CPU features which are present
+      on each of the hosts.
+     </para>
+    </step>
+   </procedure>
+
```
Comment (tbazant) [inline xml/xen_administrating.xml:297]:
```suggestion
   <tip>
    <para>
     &libvirt; also supports calculating a baseline CPU for migration. For  more details, refer to <link
xlink:href="https://documentation.suse.com/sles/15-SP2/single-html/SLES-vt-best-practices/" />.
    </para>
   </tip>
```

File: xml/xen_administrating.xml [hunk 4]
```diff
@@ -236,6 +237,82 @@
...
+     <para>
+      Start the source guest with the <emphasis>trimmed</emphasis> CPU
+      configuration. The guest can now only use CPU features which are present
+      on each of the hosts.
+     </para>
+    </step>
+   </procedure>
+   <tip>
+    <para>
+     &libvirt; also supports calculating a baseline CPU for migration. For  more details, refer to <link
```
Comment (tbazant) [inline xml/xen_administrating.xml:299]:
```suggestion
     &libvirt; also supports calculating a baseline CPU for migration. For more details, refer to <link
```

**Pull request merged**